deltacat/utils/pyarrow.py

```diff
@@ -133,6 +133,24 @@ def slice_table(
     return tables
 
 
+def slice_table_generator(
+        table: pa.Table,
+        max_len: int) -> Generator[pa.Table, None, None]:
+    """
```

Comment:
You could document this public method, and mention this wouldn't yield any results when the table is empty.